account_invoice_production_lot/tests/test_invoice_production_lot.py

```diff
@@ -78,3 +90,73 @@ def test_invoice_product_lot(self):
         self.assertEqual(len(line.prod_lot_ids.ids), 2)
         self.assertIn('Lot 1', line.lot_formatted_note)
         self.assertIn('Lot 2', line.lot_formatted_note)
+
+    def test_01_sale_stock_delivery_partial_invoice_product_lot(self):
+        # update quantities with their related lots
+        self.qty_on_hand(self.product, self.stock_location, 3, self.lot1)
+        self.qty_on_hand(self.product, self.stock_location, 3, self.lot2)
+        # confirm quotation
+        self.sale.action_confirm()
+        picking = self.sale.picking_ids[:1]
+        picking.action_confirm()
+        picking.action_assign()
+        self.assertEqual(len(picking.mapped('pack_operation_ids')), 1)
+        self.assertEqual(len(picking.mapped(
+            'pack_operation_ids.pack_lot_ids')), 2)
+        operation_lot1 = picking.pack_operation_ids.pack_lot_ids[0]
+        self.assertIn('Lot 1', operation_lot1.lot_id.name)
+        self.assertEqual(operation_lot1.qty_todo, 3)
+        operation_lot2 = picking.pack_operation_ids.pack_lot_ids[1]
+        self.assertIn('Lot 2', operation_lot2.lot_id.name)
+        self.assertEqual(operation_lot2.qty_todo, 1)
+        # deliver partially only one lot
+        operation_lot1.action_add_quantity(2)
+        picking.pack_operation_product_ids.write({'qty_done': 2})
+        backorder_wiz_id = picking.do_new_transfer()['res_id']
+        backorder_wiz = self.env['stock.backorder.confirmation'].browse(
+            [backorder_wiz_id])
+        backorder_wiz.process()
+        # create invoice
+        inv_id = self.sale.action_invoice_create()
+        invoice = self.env['account.invoice'].browse(inv_id)
+        self.assertEqual(len(invoice.invoice_line_ids), 1)
+        line = invoice.invoice_line_ids
+        # We must have only one lot
+        self.assertEqual(len(line.prod_lot_ids.ids), 1)
+        self.assertIn('Lot 1', line.lot_formatted_note)
```

Comment:
Looking at the usecase you described, here you should also assert that `'Lot 2'` is not in `line.lot_formatted_note`.
Anyway you can just skip this assertion if in the line above you check using the lot's `id`.